So the docstring test for NaT have failed but I'm not sure what's wrong: 

```
=================================== FAILURES ===================================
_____________________________ test_NaT_docstrings ______________________________

    def test_NaT_docstrings():
        # GH#17327
        nat_names = dir(NaT)
    
        # NaT should have *most* of the Timestamp methods, with matching
        # docstrings.  The attributes that are not expected to be present in NaT
        # are private methods plus `ts_expected` below.
        ts_names = dir(Timestamp)
        ts_missing = [x for x in ts_names if x not in nat_names and
                      not x.startswith('_')]
        ts_missing.sort()
        ts_expected = ['freqstr', 'normalize',
                       'to_julian_date',
                       'to_period', 'tz']
        assert ts_missing == ts_expected
    
        ts_overlap = [x for x in nat_names if x in ts_names and
                      not x.startswith('_') and
                      callable(getattr(Timestamp, x))]
        for name in ts_overlap:
            tsdoc = getattr(Timestamp, name).__doc__
            natdoc = getattr(NaT, name).__doc__
>           assert tsdoc == natdoc
E           AssertionError: assert '\n        Co...ne.\n        ' == '\n        Con...ne.\n        '
E             Skipping 383 identical leading characters in diff, use -v to show
E             -           When clocks moved backward due to DST, ambiguous times may arise.
E             -             For example in Central European Time (UTC+01), when going from 03:00 DST
E             -             to 02:00 non-DST, 02:30:00 local time occurs both at 00:30:00 UTC
E             -             and at 01:30:00 UTC. In such a situation, the `ambiguous` parameter
E             -             dictates how ambiguous times should be handled.
E             - 
E             -             - bool contains flags to determine if time is dst or not (note
E             ? --
E             +           - bool contains flags to determine if time is dst or not (note
E                             that this flag is only applicable for ambiguous fall dst dates)
E                           - 'NaT' will return NaT for an ambiguous time
E                           - 'raise' will raise an AmbiguousTimeError for an ambiguous time
E               
E                       nonexistent : 'shift', 'NaT', default 'raise'
E                           A nonexistent time does not exist in a particular timezone
E                           where clocks moved forward due to DST.
E               
E                           - 'shift' will shift the nonexistent time forward to the closest
E                             existing time
E                           - 'NaT' will return NaT where there are nonexistent times
E                           - 'raise' will raise an NonExistentTimeError if there are
E                             nonexistent times
E               
E                           .. versionadded:: 0.24.0
E               
E                       errors : 'raise', 'coerce', default None
E                           - 'raise' will raise a NonExistentTimeError if a timestamp is not
E                              valid in the specified timezone (e.g. due to a transition from
E                              or to DST time). Use ``nonexistent='raise'`` instead.
E                           - 'coerce' will return NaT if the timestamp can not be converted
E                             into the specified timezone. Use ``nonexistent='NaT'`` instead.
E               
E                             .. deprecated:: 0.24.0
E               
E                       Returns
E                       -------
E                       localized : Timestamp
E               
E                       Raises
E                       ------
E                       TypeError
E                           If the Timestamp is tz-aware and tz is not None.

pandas/tests/scalar/test_nat.py:185: AssertionError
```

--- a/pandas/_libs/tslibs/nattype.pyx
+++ b/pandas/_libs/tslibs/nattype.pyx
@@ -559,6 +559,12 @@ class NaTType(_NaT):
             None will remove timezone holding local time.
 
         ambiguous : bool, 'NaT', default 'raise'
+            When clocks moved backward due to DST, ambiguous times may arise.
+            For example in Central European Time (UTC+01), when going from 03:00 DST
+            to 02:00 non-DST, 02:30:00 local time occurs both at 00:30:00 UTC
+            and at 01:30:00 UTC. In such a situation, the `ambiguous` parameter
+            dictates how ambiguous times should be handled.
+
             - bool contains flags to determine if time is dst or not (note
               that this flag is only applicable for ambiguous fall dst dates)
             - 'NaT' will return NaT for an ambiguous time

--- a/pandas/core/arrays/datetimes.py
+++ b/pandas/core/arrays/datetimes.py
@@ -606,6 +606,11 @@ class DatetimeArrayMixin(dtl.DatetimeLikeArrayMixin):
             Time zone to convert timestamps to. Passing ``None`` will
             remove the time zone information preserving local time.
         ambiguous : 'infer', 'NaT', bool array, default 'raise'
+            When clocks moved backward due to DST, ambiguous times may arise.
+            For example in Central European Time (UTC+01), when going from 03:00 DST
+            to 02:00 non-DST, 02:30:00 local time occurs both at 00:30:00 UTC
+            and at 01:30:00 UTC. In such a situation, the `ambiguous` parameter
+            dictates how ambiguous times should be handled.
 
             - 'infer' will attempt to infer fall dst-transition hours based on
               order
@@ -677,6 +682,39 @@ class DatetimeArrayMixin(dtl.DatetimeLikeArrayMixin):
         DatetimeIndex(['2018-03-01 09:00:00', '2018-03-02 09:00:00',
                        '2018-03-03 09:00:00'],
                       dtype='datetime64[ns]', freq='D')
+
+        Be careful with DST changes. When there is sequential data, pandas can infer the DST time:
+        >>> s = pd.to_datetime(pd.Series([
+        ... '2018-10-28 01:30:00',
+        ... '2018-10-28 02:00:00',
+        ... '2018-10-28 02:30:00',
+        ... '2018-10-28 02:00:00',
+        ... '2018-10-28 02:30:00',
+        ... '2018-10-28 03:00:00',
+        ... '2018-10-28 03:30:00']))
+        >>> s.dt.tz_localize('CET', ambiguous='infer')
+        2018-10-28 01:30:00+02:00    0
+        2018-10-28 02:00:00+02:00    1
+        2018-10-28 02:30:00+02:00    2
+        2018-10-28 02:00:00+01:00    3
+        2018-10-28 02:30:00+01:00    4
+        2018-10-28 03:00:00+01:00    5
+        2018-10-28 03:30:00+01:00    6
+        dtype: int64
+
+        In some cases, inferring the DST is impossible. In such cases, you can
+        pass an ndarray to the ambiguous parameter to set the DST explicitly
+
+        >>> s = pd.to_datetime(pd.Series([
+        ... '2018-10-28 01:20:00',
+        ... '2018-10-28 02:36:00',
+        ... '2018-10-28 03:46:00']))
+        >>> s.dt.tz_localize('CET', ambiguous=np.array([True, True, False]))
+        0   2018-10-28 01:20:00+02:00
+        1   2018-10-28 02:36:00+02:00
+        2   2018-10-28 03:46:00+01:00
+        dtype: datetime64[ns, CET]
+
         """
         if errors is not None:
             warnings.warn("The errors argument is deprecated and will be "

--- a/pandas/core/generic.py
+++ b/pandas/core/generic.py
@@ -8662,14 +8662,14 @@ class NDFrame(PandasObject, SelectionMixin):
 
         Localize local times:
 
-        >>> s = pd.to_datetime(pd.Series(['2018-09-15 01:30:00']))
-        >>> s.dt.tz_localize('CET')
-        0   2018-09-15 01:30:00+02:00
-        dtype: datetime64[ns, CET]
+        >>> s = pd.Series([1], index=pd.DatetimeIndex(['2018-09-15 01:30:00']))
+        >>> s.tz_localize('CET')
+        2018-09-15 01:30:00+02:00    1
+        dtype: int64
 
         Be careful with DST changes. When there is sequential data, pandas can infer the DST time:
 
-        >>> s = pd.to_datetime(pd.Series([
+        >>> s = pd.Series(range(7), index=pd.DatetimeIndex([
         ... '2018-10-28 01:30:00',
         ... '2018-10-28 02:00:00',
         ... '2018-10-28 02:30:00',
@@ -8678,27 +8678,27 @@ class NDFrame(PandasObject, SelectionMixin):
         ... '2018-10-28 03:00:00',
         ... '2018-10-28 03:30:00']))
         >>> s.dt.tz_localize('CET', ambiguous='infer')
-        1   2018-10-28 01:30:00+02:00
-        2   2018-10-28 02:00:00+02:00
-        3   2018-10-28 02:30:00+02:00
-        4   2018-10-28 02:00:00+01:00
-        5   2018-10-28 02:30:00+01:00
-        6   2018-10-28 03:00:00+01:00
-        7   2018-10-28 03:30:00+01:00
-        dtype: datetime64[ns, CET]
+        2018-10-28 01:30:00+02:00    0
+        2018-10-28 02:00:00+02:00    1
+        2018-10-28 02:30:00+02:00    2
+        2018-10-28 02:00:00+01:00    3
+        2018-10-28 02:30:00+01:00    4
+        2018-10-28 03:00:00+01:00    5
+        2018-10-28 03:30:00+01:00    6
+        dtype: int64
 
         In some cases, inferring the DST is impossible. In such cases, you can
         pass an ndarray to the ambiguous parameter to set the DST explicitly
 
-        >>> s = pd.to_datetime(pd.Series([
+        >>> s = pd.Series(range(3), index=pd.DatetimeIndex([
         ... '2018-10-28 01:20:00',
         ... '2018-10-28 02:36:00',
         ... '2018-10-28 03:46:00']))
-        >>> s.dt.tz_localize('CET', ambiguous=np.array([True, True, False]))
-        0   2018-10-28 01:20:00+02:00
-        1   2018-10-28 02:36:00+02:00
-        2   2018-10-28 03:46:00+01:00
-        dtype: datetime64[ns, CET]
+        >>> s.tz_localize('CET', ambiguous=np.array([True, True, False]))
+        2018-10-28 01:20:00+02:00    0
+        2018-10-28 02:36:00+02:00    1
+        2018-10-28 03:46:00+01:00    2
+        dtype: int64
 
         """
         if nonexistent not in ('raise', 'NaT', 'shift'):

--- a/pandas/core/indexes/datetimes.py
+++ b/pandas/core/indexes/datetimes.py
@@ -180,44 +180,6 @@ class DatetimeIndex(DatetimeArrayMixin, DatelikeOps, TimelikeOps,
     PeriodIndex : Index of Period data
     pandas.to_datetime : Convert argument to datetime
 
-    Examples
-    --------
-
-        Localize local times:
-
-        >>> s = pd.to_datetime(pd.Series(['2018-09-15 01:30:00']))
-        >>> pd.DatetimeIndex(s, tz='CET')
-        DatetimeIndex(['2018-09-15 01:30:00+02:00'], dtype='datetime64[ns, CET]', freq=None)
-
-        Be careful with DST changes. When there is sequential data, pandas can infer the DST time:
-
-        >>> s = pd.to_datetime(pd.Series([
-        ... '2018-10-28 01:30:00',
-        ... '2018-10-28 02:00:00',
-        ... '2018-10-28 02:30:00',
-        ... '2018-10-28 02:00:00',
-        ... '2018-10-28 02:30:00',
-        ... '2018-10-28 03:00:00',
-        ... '2018-10-28 03:30:00']))
-        >>> pd.DatetimeIndex(s, tz='CET', ambiguous='infer')
-        DatetimeIndex(['2018-10-28 01:30:00+02:00', '2018-10-28 02:00:00+02:00',
-                       '2018-10-28 02:30:00+02:00', '2018-10-28 02:00:00+01:00',
-                       '2018-10-28 02:30:00+01:00', '2018-10-28 03:00:00+01:00',
-                       '2018-10-28 03:30:00+01:00'],
-                      dtype='datetime64[ns, CET]', freq=None)
-
-        In some cases, inferring the DST is impossible. In such cases, you can
-        pass an ndarray to the ambiguous parameter to set the DST explicitly
-
-        >>> s = pd.to_datetime(pd.Series([
-        ... '2018-10-28 01:20:00',
-        ... '2018-10-28 02:36:00',
-        ... '2018-10-28 03:46:00']))
-        >>> pd.DatetimeIndex(s, tz='CET', ambiguous=np.array([True, True, False]))
-        DatetimeIndex(['2018-10-28 01:20:00+02:00', '2018-10-28 02:36:00+02:00',
-                       '2018-10-28 03:46:00+01:00'],
-                      dtype='datetime64[ns, CET]', freq=None)
-
     """
     _resolution = cache_readonly(DatetimeArrayMixin._resolution.fget)
 

